webapp/src/main/java/org/iqu/webapp/factory/ServiceFactory.java

```diff
@@ -20,105 +20,105 @@
-	}
+  /**
+   * This method is used to ensure the return of the same instance of
+   * EventsServiceSlave for all threads.
+   */
+  public static synchronized EventsServiceSlave getEventsServiceInstance() {
+    if (eventsServiceInstance == null) {
+      synchronized (ServiceFactory.class) {
+        if (eventsServiceInstance == null) {
+          eventsServiceInstance = new EventsServiceSlaveImpl();
+        }
+      }
+    }
+    return eventsServiceInstance;
+  }
 
-	/**
-	 * This method is used to ensure the return of the same instance of
-	 * NewsServiceSlave for all threads.
-	 */
-	public static synchronized NewsServiceSlave getNewsServiceInstance() {
-		if (newsServiceInstance == null) {
-			synchronized (ServiceFactory.class) {
-				if (newsServiceInstance == null) {
-					newsServiceInstance = new NewsServiceSlaveImpl();
-				}
-			}
-		}
-		return newsServiceInstance;
-	}
+  /**
+   * This method is used to ensure the return of the same instance of
+   * NewsServiceSlave for all threads.
+   */
+  public static synchronized NewsServiceSlave getNewsServiceInstance() {
+    if (newsServiceInstance == null) {
+      synchronized (ServiceFactory.class) {
+        if (newsServiceInstance == null) {
+          newsServiceInstance = new NewsServiceSlaveImpl();
+        }
+      }
+    }
+    return newsServiceInstance;
+  }
 
-	/**
-	 * This method is used to ensure the return of the same instance of
-	 * HealthCheckServiceSlave for all threads.
-	 */
-	public static synchronized HealthCheckServiceSlave getOtherServiceInstance() {
-		if (healthCheckServiceInstance == null) {
-			synchronized (ServiceFactory.class) {
-				if (healthCheckServiceInstance == null) {
-					healthCheckServiceInstance = new HealthCheckServiceSlaveImpl();
-				}
-			}
-		}
-		return healthCheckServiceInstance;
-	}
+  /**
+   * This method is used to ensure the return of the same instance of
+   * HealthCheckServiceSlave for all threads.
+   */
+  public static synchronized HealthCheckServiceSlave getOtherServiceInstance() {
+    if (healthCheckServiceInstance == null) {
+      synchronized (ServiceFactory.class) {
+        if (healthCheckServiceInstance == null) {
+          healthCheckServiceInstance = new HealthCheckServiceSlaveImpl();
+        }
+      }
+    }
+    return healthCheckServiceInstance;
+  }
 
-	/**
-	 * This method is used to ensure the return of the same instance of
-	 * LocationServiceSlave for all threads.
-	 */
-	public static synchronized LocationServiceSlave getLocationServiceInstance() {
-		if (locationServiceInstance == null) {
-			synchronized (ServiceFactory.class) {
-				if (locationServiceInstance == null) {
-					locationServiceInstance = new LocationServiceSlaveImpl();
-				}
-			}
-		}
-		return locationServiceInstance;
-	}
+  /**
+   * This method is used to ensure the return of the same instance of
+   * LocationServiceSlave for all threads.
+   */
+  public static synchronized LocationServiceSlave getLocationServiceInstance() {
```

Comment:
please remove the synchronized keyword from the method... you already are guarding against race condition in the body of function